core/chains/evm/txmgr_wrapper.go

```diff
@@ -0,0 +1,117 @@
+package evm
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/smartcontractkit/sqlx"
+
+	txmgrtypes "github.com/smartcontractkit/chainlink/common/txmgr/types"
+	evmclient "github.com/smartcontractkit/chainlink/core/chains/evm/client"
+	evmconfig "github.com/smartcontractkit/chainlink/core/chains/evm/config"
+	httypes "github.com/smartcontractkit/chainlink/core/chains/evm/headtracker/types"
+	"github.com/smartcontractkit/chainlink/core/chains/evm/logpoller"
+	"github.com/smartcontractkit/chainlink/core/chains/evm/txmgr"
+	evmtypes "github.com/smartcontractkit/chainlink/core/chains/evm/types"
+	"github.com/smartcontractkit/chainlink/core/logger"
+	"github.com/smartcontractkit/chainlink/core/services"
+	"github.com/smartcontractkit/chainlink/core/utils"
+)
+
+var _ httypes.HeadTrackable = &txmWrapper{}
+
+// EVM specific wrapper to hold the core TxMgr object underneath
+type txmWrapper struct {
```

Comment:
What's the reasoning behind introducing this wrapper in this PR? Is it necessary to make Head Tracker generic?

Reply:
The TXM, after the generic HeadTracking changes implements the following trackable:
OnNewLongestChain(ctx context.Context, head txmgrtypes.HeadView)

Earlier, the last parameter was *evmtypes.Head. It was evm specific, but now is generic. 
However, the HeadBroadcaster, which calls this method, still expects the evmtypes.Head param. Rather than change the broadcaster, and thus all subscribers across the codebase, I created a txmWrapper, which accepts evmtypes.Head, and then converts that to HeadView, and then calls the Txm's implementation.
Thus an easy hack to make the TXM generic

Reply:
can we call this struct `EvmTxm` since it will become EVM specific?

and then it would be `NewEvmTxManager`. we will need to use it to construct the chain-specific gas estimators, etc before passing it to the chain-agnostic TXM

and then the struct would be something like
```go
type EvmTxm struct {
  txmgr.TxManager
  httypes.HeadTrackable
}

func (txm *EvmTxm) OnNewLongestChain(ctx context.Context, evmHead *evmtypes.Head) { ... }
```

this way you don't need to redefine a bunch of interfaces because the `TxManager` interface already has them, and all you need is to wrap a function